include/gul14/traits.h

```diff
@@ -15,6 +15,7 @@
 #ifndef GUL14_TRAITS_H_
 #define GUL14_TRAITS_H_
 
+#include <functional> // for std::invoke on C++17 compilers
```

Comment:
```suggestion
#if __cplusplus >= 201703L
#include <functional> // for std::invoke on C++17 compilers
#endif
```

All in all I would put that into a new macro somewhere in the file top and reuse that local macro everywhere we switch between `std` and `gul14` in the file.

Reply:
> All in all I would put that into a new macro somewhere in the file top and reuse that local macro everywhere we switch between std and gul14 in the file.

I tried that in 4e9444d22cb6a6409260ec447dd06034d1ddc48b, but I do not like the result. It adds another macro and complicates the code just to shave a few milliseconds off of someone's build times. I'd vote for just using the first commit.

Reply:
In fact, I'll cherry-pick the first commit into `main`. The discussion can continue here.

Reply:
Is there a reason we use `std::invoke` if we are on C++17?
I believe we use `gul::string_view` on 17 even though `std::string_view` is available?